### DISCUSSION TITLE
Differences in the printer.cfg file vs official klipper example (SV06 plus)

### DISCUSSION THREAD

**Discussion by rocket59** [bassamanator/Sovol-SV06-firmware#165] [General]
I was comparing the printer cfg files from the klipper github against the @bassamanator one and noticed some differences, the main one being 16 microstepping in the klipper one but 256 on the bassamanator. Also the tmc2209 currents are different as well as extruder rotation, bed mesh settings, and no screws tilt adjust settings nor axis twist compensation in the klipper version.

I'm setting it up on a Sonic Pad and appreciate that its version of klipper is well behind. Was  curious as to the rationale for the differences?

**Comment by bassamanator** [comment]
> 256 ??
>  tmc2209 currents ??

I think you're using an old copy of my config. In any case, there are differences.

My `printer.cfg` is a fork of [@spinixguy](https://github.com/spinixguy/Sovol-SV06-firmware)'s (he's the one who submitted his `printer.cfg` to the klipper repo). The klipper `printer.cfg` is not the best version of itself, it's only the base, and in some case it's incorrect according to their own guidelines, I've submitted corrections myself (look for my name in their commit history). My `printer.cfg` is more heavily tested and incorporates new additions to klipper, whereas the klipper version is from 2022. Also check the number of commits, albeit a small indicator of work done and testing, but an indicator nonetheless. You'll find Sovol official `printer.cfg` had different values compared to Sovol marlin source code as well.

I've made changes that I think are best. Use whatever you think is best.

**Comment by rocket59** [comment]
Thanks for the reply and all the work you've put in! I didn't did not mean to cause any offence, and if I did I totally apologise. 

Just had another look at the files and yours is 128 microstepping (honestly I don't recall what I was looking at before). I'll go through the commits in more detail to satisfy my curiosity but the tmc2209 run_currents are different (albeit slightly - it might be worth putting in a change request?)

**Comment by bassamanator** [comment]
Not offended at all, my friend. 

Please link to the lines of code, the run_current, that are different because I don't see it. In any case, it really doesn't matter, because this is not an exact science, I know that sounds strange. As I mentioned, I've noticed discrepancies between Sovol offical klipper settings and what's in their marlin source code. For example, the currents can vary by roughly 0.1 amps, or more, without issue.

**Comment by rocket59** [comment]
From the klipper config here https://github.com/Klipper3d/klipper/blob/master/config/printer-sovol-sv06-plus-2023.cfg

As an example, from this file: (line 37)
```
[tmc2209 stepper_x]
uart_pin: PC1
run_current: 0.860
sense_resistor: 0.150
uart_address: 3
driver_SGTHRS: 86
diag_pin: PA5
```

From yours:
```
[tmc2209 stepper_x]
uart_pin: PC1
run_current: 0.820
stealthchop_threshold: 0
interpolate: False
sense_resistor: 0.150
uart_address: 3
driver_SGTHRS: 86 # NOTE Bump sensitivity; increase to make more sensitive
diag_pin: PA5
```

As you point out, only a minor difference which pretty much makes no difference in real life (literally a touch of the adjust screw in an 8825 :)).

**Comment by bassamanator** [comment]
So this is for the `plus`. Yes the run_current is different, however, mine matches what's in the marlin source code, these other values I'm not sure about. They could be 'wrong' or perhaps they match Sovol official `printer.cfg`? Have a look and let us all know if you're interested to do so. You'll find this info [here](https://github.com/Sovol3d/SOVOL_KLIPAD50_SYSTEM?tab=readme-ov-file).

**Comment by rocket59** [comment]
Thanks for the link. I'll have a look for the differences.
